packages/kit/src/runtime/server/page/fetch.js

```diff
@@ -68,6 +68,13 @@ export function create_fetch({ event, options, state, route, prerender_default,
 
 				const url = new URL(request.url);
 
+				if (
+					!request.headers.get('origin') &&
```

Comment:
In https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#description it says "if a cross-origin GET or HEAD request is made in no-cors mode, the Origin header will not be added" - should we add that check here, or is this irrelevant here?

```suggestion
					!request.headers.get('origin') &&
					!request.mode !== 'no-cors' &&
```

Reply:
Hm, I think it's a good idea to stick as close to the spec as possible if the goal is to simulate browser behaviour. In that case we'd also have to add the `Origin` to all *same-origin* requests that are **not** `GET` or `HEAD` and exclude from `GET` and `HEAD` if the request is "no-cors"

Reply:
That is already part of the if-condition

Reply:
Is it? It doesn't check `HEAD` (not sure if relevant) and the `no-cors` check would omit the `Origin` header for all no-cors requests (but should only omit for `GET` and `HEAD`).

Reply:
My proposition to solve this:
```js
if (!request.headers.has('origin')) {
	request.headers.set('origin', event.url.origin);
}

// Remove Origin, according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#description
if (
	(request.method === 'GET' || request.method === 'HEAD') &&
	((request.mode === 'no-cors' && url.origin !== event.url.origin) || url.origin === event.url.origin)
) {
	request.headers.delete('origin');
}
```

What do you think @dummdidumm ? This way the origin is always added, but removed for the special cases outlined on MDN where the origin is omitted

Reply:
I have modified the code with the changes outlined above. This should ensure the same behaviour as is described on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#description). Eg.
- Add `Origin` to all cross-origin requests
- Don't send the `Origin` header for **same-origin** `GET` and `HEAD` requests
- Don't send the `Origin` header for **cross-origin** `"no-cors"` `GET` and `HEAD` requests